src/c-writer.cc

```diff
@@ -2351,7 +2351,11 @@ void CWriter::WriteFuncRefWrapper(const Func* func) {
   Write(") ", OpenBrace());
   Write("return ");
   Write(ExternalRef(ModuleFieldType::Func, func->name));
-  Write("(instance");
+  std::string target_module_name =
+      func->ext_module_name == ""
+          ? ModuleInstanceTypeName()
+          : ModuleInstanceTypeName(func->ext_module_name);
+  Write("((struct ", target_module_name, "*) instance");
```

Comment:
Can we not somehow do dynamic check here and downcast to a `FuncImport`?   I guess we don't have the RTTI info in the IR classes to do a downcast?   

It looks like we have `IsImport` which can check if the function is imported.. and then do a cast mabye?

Reply:
Yup, I am using `IsImport` to figure this out now.